src/wp-includes/blocks/post-terms.php

```diff
@@ -102,12 +104,18 @@ function register_block_core_post_terms() {
 			$custom_variations[] = $variation;
 		}
 	}
+	return array_merge( $built_ins, $custom_variations );
+}
 
+/**
+ * Registers the `core/post-terms` block on the server.
+ */
+function register_block_core_post_terms() {
 	register_block_type_from_metadata(
 		__DIR__ . '/post-terms',
 		array(
 			'render_callback' => 'render_block_core_post_terms',
-			'variations'      => array_merge( $built_ins, $custom_variations ),
+			'variations'      => 'get_variations_core_post_terms',
```

Comment:
It would be nice to use a consistent naming scheme for all of the variation callbacks in core blocks. Right now you have:

* get_variations_core_navigation_link
* get_variations_core_navigation_link
* build_template_part_block_variations

Personally, I'd go with something like `get_block_core_{block_name}_variations` which is similar to the consistent render callback `render_block_core_{block_name}`.

Reply:
I was avoiding `build_{block name}` as the navigation link already had a similar function with a different purpose ->`build_variation_for_navigation_link`.

I like the `get_block_core_{block_name}_variations` naming throughout, but we may have to deprecate the function `build_template_part_block_variations` which is old and can be in use in ways we don't know.
For now, I have made everything `build_{block}_block_variations` for consistency.

Reply:
@joemcgill  I have created a draft PR in GB repo - https://github.com/WordPress/gutenberg/pull/56952/files

There are some CS issues which I think someone from GB would be the best person to address.